mkfs/dump.c

```diff
@@ -98,6 +102,49 @@ closure_function(3, 2, void, fsc,
     readdir(fs, h, bound(root), bound(b));
 }
 
+static void print_colored(int indent, int color, symbol s, boolean newline)
+{
+    while (indent--) {
+        console("|");
+        for (int i = 1; i < 4; i++)
+            console(" ");
+    }
+    buffer tmpbuf = little_stack_buffer(NAME_MAX + 1);
+    printf("\e[%dm%s\e[%dm%s", color, cstring(symbol_string(s), tmpbuf),
+           TERM_COLOR_WHITE, newline ? "\n" : "");
+}
+
+static void dump_fsentry(int indent, symbol name, tuple t)
+{
+    table c;
+    buffer target_buf;
+    if ((c = children(t))) {
+        print_colored(indent, TERM_COLOR_BLUE, name, true);
+        table_foreach((tuple)c, k, vc) {
+            if (k == sym_this(".") || k == sym_this(".."))
```

Comment:
Not in the scope of this PR, but it would be really nice if we could do away with these links. I realize the ascending link obviates the need for a traversal, but it just seems icky to have these hanging around at every node - and persistently on storage.

Reply:
Probably it won't make this any less icky, but the . and .. directories entries are not persisted on storage (see `fs_set_dir_entry()` in tfs.c), they are created at runtime when the filesystem is read, by the `fixup_directory()` function.